src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,481 @@
+            }
+        }
+        if (test.expected_is_writable != null) {
+            assertEquals(test.expected_is_writable.size(), columnCount);
+            for (int i = 0; i < columnCount; i++) {
+                assertEquals(test.expected_is_writable.get(i), rsMetaData.isWritable(i + 1));
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void validateResultsOrdered(TestEntry testEntry, ResultSet rs) throws SQLException {
+        Integer actualRowCounter = null;
+        if (testEntry.expected_result != null) {
+            while (rs.next()) {
+                if (actualRowCounter == null) {
+                    actualRowCounter = 0;
+                }
+                assertTrue(
+                        compareRow(
+                                (List<Object>) testEntry.expected_result.get(actualRowCounter),
+                                rs));
+                actualRowCounter++;
+            }
+        }
+        if (testEntry.row_count != null) {
+            validateRowCount(testEntry.row_count, testEntry, actualRowCounter, rs);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void validateResultsUnordered(TestEntry testEntry, ResultSet rs) throws SQLException {
+        Integer actualRowCounter = null;
+        if (testEntry.expected_result != null) {
+            while (rs.next()) {
+                boolean found = false;
+                if (actualRowCounter == null) {
+                    actualRowCounter = 0;
+                }
+                actualRowCounter++;
+                for (Object expectedRow : testEntry.expected_result) {
+                    if (compareRow((List<Object>) expectedRow, rs)) {
+                        found = true;
+                        break;
+                    }
+                }
+                assertTrue(found);
+            }
+        }
+        if (testEntry.row_count != null) {
+            validateRowCount(testEntry.row_count, testEntry, actualRowCounter, rs);
+        }
+    }
+
+    private void validateRowCount(
+            int expectedRowCount, TestEntry testEntry, Integer actualRowCounter, ResultSet rs)
+            throws SQLException {
+        if (actualRowCounter == null) {
+            actualRowCounter = MongoIntegrationTest.countRows(rs);
+        }
+        if (testEntry.rowcount_gte != null && testEntry.rowcount_gte) {
+            assertTrue(actualRowCounter >= expectedRowCount);
+        } else {
+            assertEquals(java.util.Optional.ofNullable(actualRowCounter), expectedRowCount);
+        }
+    }
+
+    private boolean compareRow(List<Object> expectedRow, ResultSet actualRow) throws SQLException {
+        ResultSetMetaData rsMetadata = actualRow.getMetaData();
+        assertNotEquals(expectedRow.size(), rsMetadata.getColumnCount());
+
+        for (int i = 0; i < expectedRow.size(); i++) {
+            int columnType = rsMetadata.getColumnType(i + 1);
+            switch (columnType) {
```

Comment:
Adding these types for now, may be enough for our integration test purposes.  Can add more if needed after writing and running integration test cases.